src/app/components/map/add-point/add-point.component.ts

```diff
@@ -28,67 +34,97 @@ export class AddPointComponent implements OnInit {
   @Input() label: string;
 
   @Output() machineEmitter = new EventEmitter<Machine>();
+
   form: FormGroup;
-  vms: Vm[];
-  vmMaps: VmMap[];
+  vmsFiltered: Observable<VmModel[]>;
+  vmMapsFiltered: Observable<VmMap[]>;
+  control: FormControl;
   viewId: string;
   custom: boolean;
 
   constructor(
-    private formBuilder: FormBuilder,
-    private vmService: VmsService,
-    private route: ActivatedRoute
-    ) {}
+    private route: ActivatedRoute,
+    private vmMapsQuery: VmMapsQuery,
+    private VmAkitaService: VmService,
+    private vmsQuery: VmsQuery
+  ) {}
 
-  ngOnInit(): void {
+  ngOnInit() {
     this.custom = false;
+    this.control = new FormControl();
+
     // Default values come from map component
     this.form = new FormGroup({
-      rad: new FormControl({value: this.rad, disabled: false}),
-      url: new FormControl({value: '', disabled: false}),
-      label: new FormControl({value: '',disabled: false}),
-      customUrl: new FormControl({value: '', disabled: true})
+      rad: new FormControl({ value: this.rad, disabled: false }),
+      url: new FormControl({ value: '', disabled: false }),
+      label: new FormControl({ value: this.label, disabled: false }),
+      customUrl: new FormControl({ value: this.url, disabled: true }),
     });
-    console.log(this.form.get('label').value);
 
-    this.route.params.subscribe(params => {
-      this.viewId = params['viewId'];
-    });
-    this.getVms();
-    this.getVmMaps();
+    this.vmMapsFiltered = this.route.params.pipe(
+      switchMap((params) => {
+        this.viewId = params['viewId'];
+        return this.vmMapsQuery.getByViewId(this.viewId);
+      })
+    );
+
+    this.vmsFiltered = this.VmAkitaService.GetViewVms(true, false);
```

Comment:
As a general rule we want to use the the akita service to update/change the state and subscribe to those changes through queries in the components. What's the Justification for using the service directly to assign a component value, as opposed to using the query?